code/general/membership.exs

```diff
@@ -0,0 +1,64 @@
+defmodule Membership.NewMapSetMember do
+  def check_membership(range, list, _mapset) do
+    find = Enum.random(range)
+    mapset = MapSet.new(list)
+    MapSet.member?(mapset, find)
+  end
+end
+
+defmodule Membership.MapSetMember do
+  def check_membership(range, _list, mapset) do
+    find = Enum.random(range)
+    MapSet.member?(mapset, find)
+  end
+end
+
+defmodule Membership.EnumMember do
+  def check_membership(range, list, _mapset) do
+    find = Enum.random(range)
+    Enum.member?(list, find)
+  end
+end
+
+defmodule Membership.EnumAny do
+  def check_membership(range, list, _mapset) do
+    find = Enum.random(range)
+    Enum.any?(list, &(&1 == find))
+  end
+end
+
+defmodule Membership.In do
+  def check_membership(range, list, _mapset) do
+    find = Enum.random(range)
+    find in list
+  end
+end
+
+defmodule Membership.Benchmark do
+  @inputs %{
+    "Large" => {1..1_000_000, Enum.shuffle(1..1_000_000), MapSet.new(1..1_000_000)},
+    "Medium" => {1..10_000, Enum.shuffle(1..10_000), MapSet.new(1..10_000)},
+    "Small" => {1..100, Enum.shuffle(1..100), MapSet.new(1..100)}
+  }
+
+  def benchmark do
+    Benchee.run(
+      %{
+        "New MapSet + MapSet.member?" => fn input -> bench(Membership.NewMapSetMember, input) end,
```

Comment:
Wasn't sure whether to keep this scenario. It ends up being the slowest for all inputs. Including may be useful to show that you lose all performance benefits if you're creating a new mapset that gets infrequently used.

Reply:
I'm _very_ in favor of keeping this. Sometimes when folks look at stuff like this they think "oh, MapSet is faster, so I'll use that!" but in the end the added runtime of the conversion ends up making their code slower overall.

That's the unfortunate consequence of microbenchmarks. They can be helpful, but you also need to look at the macro picture.